### ISSUE TITLE
Ability to disable scan file limit in FileManager

### ISSUE THREAD

**Issue by milleruntime** [apache/accumulo#1606]
It seems the lock contention due to the file limit in FileManager is still an issue in 1.9.  This problem has been around since https://issues.apache.org/jira/browse/ACCUMULO-416 which is a prerequisite for https://issues.apache.org/jira/browse/ACCUMULO-543.  It would be worthwhile to revisit the first ticket and at the very least have a config option to disable the file limit lock.

**Comment by milleruntime** [comment]
Saw an issue where short scans were locking at this file limit gate so the client was killed.  The dead scans were stuck even after a client was killed and then new scans were waiting a long time at the semaphore for this file limit.  This situation could worsen as a user not seeing scans available, will retry creating more scans that get held up at this gate.   This is a situation where we want to prioritize scan availability over everything else, which is one of the trade offs @keith-turner mentions in https://issues.apache.org/jira/browse/ACCUMULO-416

**Comment by ctubbsii** [comment]
Isn't the number of open files configurable? It seems like increasing that limit would mitigate this a bit.

**Comment by milleruntime** [comment]
> Isn't the number of open files configurable? It seems like increasing that limit would mitigate this a bit.

Yes there is "tserver.scan.files.open.max".  I suggested increasing this but now now sure why this wouldn't suffice.  @FineAndDandy any idea why just increasing this limit wouldn't be enough to bypass this gate where necessary?

**Comment by dlmarion** [comment]
@FineAndDandy  - any updated information on the question in the comment above? Is this still an issue?

**Comment by dlmarion** [comment]
I talked with @FineAndDandy today, this is still an issue. They are still having to increase `tserver.scan.files.open.max` when they run into this issue.

**Comment by ctubbsii** [comment]
Why is that configuration property not sufficient to address this? Is something else needed, and if so, why?
